concrete-core-fixture/src/fixture/glwe_ciphertext_decryption.rs

```diff
@@ -0,0 +1,148 @@
+use crate::fixture::Fixture;
+use crate::generation::prototyping::{
+    PrototypesGlweCiphertext, PrototypesGlweSecretKey, PrototypesPlaintextVector,
+};
+use crate::generation::synthesizing::{
+    SynthesizesGlweCiphertext, SynthesizesGlweSecretKey, SynthesizesPlaintextVector,
+};
+use crate::generation::{IntegerPrecision, Maker};
+use crate::raw::generation::RawUnsignedIntegers;
+use crate::raw::statistical_test::assert_noise_distribution;
+use concrete_commons::dispersion::Variance;
+use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+use concrete_core::prelude::{
+    GlweCiphertextDecryptionEngine, GlweCiphertextEntity, GlweSecretKeyEntity,
+    PlaintextVectorEntity,
+};
+
+/// A fixture for the types implementing the `GlweCiphertextDecryptionEngine` trait.
+pub struct GlweCiphertextDecryptionFixture;
+
+#[derive(Debug)]
+pub struct GlweCiphertextDecryptionParameters {
+    pub noise: Variance,
+    pub glwe_dimension: GlweDimension,
+    pub polynomial_size: PolynomialSize,
+}
+
+impl<Precision, Engine, PlaintextVector, SecretKey, Ciphertext>
+    Fixture<Precision, Engine, (PlaintextVector, SecretKey, Ciphertext)>
+    for GlweCiphertextDecryptionFixture
+where
+    Precision: IntegerPrecision,
+    Engine: GlweCiphertextDecryptionEngine<SecretKey, Ciphertext, PlaintextVector>,
+    PlaintextVector: PlaintextVectorEntity,
+    SecretKey: GlweSecretKeyEntity,
+    Ciphertext: GlweCiphertextEntity<KeyDistribution = SecretKey::KeyDistribution>,
+    Maker: SynthesizesPlaintextVector<Precision, PlaintextVector>
+        + SynthesizesGlweSecretKey<Precision, SecretKey>
+        + SynthesizesGlweCiphertext<Precision, Ciphertext>,
+{
+    type Parameters = GlweCiphertextDecryptionParameters;
+    type RepetitionPrototypes = (
+        <Maker as PrototypesGlweSecretKey<Precision, SecretKey::KeyDistribution>>::GlweSecretKeyProto,
+    );
+    type SamplePrototypes = (
+        <Maker as PrototypesPlaintextVector<Precision>>::PlaintextVectorProto,
+        <Maker as PrototypesGlweCiphertext<Precision, SecretKey::KeyDistribution>>::GlweCiphertextProto,
+    );
+    type PreExecutionContext = (SecretKey, Ciphertext);
+    type PostExecutionContext = (SecretKey, Ciphertext, PlaintextVector);
+    type Criteria = (Variance,);
+    type Outcome = (Vec<Precision::Raw>, Vec<Precision::Raw>);
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![GlweCiphertextDecryptionParameters {
+                noise: Variance(0.00000001),
+                glwe_dimension: GlweDimension(200),
```

Comment:
Here we should also be testing for a GLWE dimension of 1.

Reply:
Just did ! Thanks !